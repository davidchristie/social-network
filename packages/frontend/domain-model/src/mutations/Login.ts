import gql from "graphql-tag";

const Login = gql`
  mutation Login($email: String! $password: String!) {
    login(email: $email, password: $password) {
      token
    }
  }
`;

export default Login;
