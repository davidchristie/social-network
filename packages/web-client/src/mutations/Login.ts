import gql from "graphql-tag";

export interface LoginData {
  login: {
    token: string;
  };
}

export interface LoginVariables {
  email: string;
  password: string;
}

export default gql`
  mutation Login($email: String! $password: String!) {
    login(email: $email, password: $password) {
      token
    }
  }
`;
