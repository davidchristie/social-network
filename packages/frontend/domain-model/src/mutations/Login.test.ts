import Login from "./Login";

describe("Login mutation", () => {
  it("matches snapshot", () => {
    expect(Login).toMatchSnapshot();
  });
});
