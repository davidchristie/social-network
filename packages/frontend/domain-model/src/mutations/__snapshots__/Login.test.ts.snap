// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login mutation matches snapshot 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "Login",
      },
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "email",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "email",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "password",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "password",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "login",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "token",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "email",
            },
          },
        },
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "password",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 126,
    "start": 0,
  },
}
`;
